### DISCUSSION TITLE
Carbon Components CSS classes documentation

### DISCUSSION THREAD

**Discussion by nSedrickm** [carbon-design-system/carbon#8527] [Help]
Hello, Greetings 

Please is there documentation for Carbon CSS classes like that of [bootstrap](https://getbootstrap.com/)?

I have been searching for a way to use carbon styles with [styled-components](https://styled-components.com/) to build custom components

**Comment by jnm2377** [comment]
Hey @nSedrickm unfortunately we don't have a doc that shows all our css classes. I'm not sure how helpful this would be, but I'd recommend looking at the internal markup of our react components so you can see the elements used and the classes assigned to them. Again, Idk if this is the best strategy. I wonder if @lee-chase maybe could give some input here on how the vue maintainers keep up with classname changes and stuff.

**Reply by lee-chase** [reply]
I don't know very much about styled-components, but a brief look appears to require the styles to be read in JS. Carbon styles are currently all SCSS with files like this one showing some of the structure and how each component constructs its styles. https://github.com/carbon-design-system/carbon/blob/main/packages/components/src/globals/scss/styles.scss

You may well know more than the Carbon team does about moving from SCSS to styled-components.
